lib/cocoapods/installer/target_installer/pod_target_installer.rb

```diff
@@ -22,15 +22,13 @@ def install!
           create_xcconfig_file
           if target.requires_frameworks?
             create_info_plist_file
+            create_module_map do |generator|
+              generator.private_headers += target.file_accessors.flat_map(&:private_headers).map(&:basename)
+            end
+            create_umbrella_header do |generator|
+              generator.imports += target.file_accessors.flat_map(&:public_headers).map(&:basename)
+            end
           end
-          create_module_map do |generator|
-            generator.private_headers += target.file_accessors.flat_map(&:private_headers).map(&:basename)
-          end
-          create_umbrella_header do |generator|
-            generator.imports += target.file_accessors.flat_map(&:public_headers).map(&:basename)
-          end
-          link_module_map
```

Comment:
Got it. Yep, I can see at least one reason why the Realm pod broke: it finds both the original modulemap and the module map in Headers/Public. :(